README.md
<!--
 * @Author: xingjin
 * @Date: 2023-07-21 15:39:26
 * @LastEditors: xingjinjin
 * @LastEditTime: 2023-09-05 09:32:48
 * @Description: 请填写简介
-->
# my-blog

#### 介绍
博客管理平台使用strapi 搭建 

#### 软件架构
软件架构说明


#### 安装教程

1.  pnpm i
2.  pnpm dev
3.  

#### 使用说明

1.  提交时有提交校验
2.  代码经格式化后再进行提交（自动格式化以及eslint错误提示，禁止直接使用--fore强推）
3.  代码直接部署到github pages 链接是 [我的个人博客](https://besthandsome.github.io/vue3-blog/#/home)
4.  组件库使用naive ui 还有vue vue-router pinia 均已添加自动导入页面无需二次引入

#### 参与贡献

1.  Fork 本仓库
2.  新建 Feat_xxx 分支
3.  提交代码
4.  新建 Pull Request


#### 特技

1.  使用 Readme\_XXX.md 来支持不同的语言，例如 Readme\_en.md, Readme\_zh.md
2.  Gitee 官方博客 [blog.gitee.com](https://blog.gitee.com)
3.  你可以 [https://gitee.com/explore](https://gitee.com/explore) 这个地址来了解 Gitee 上的优秀开源项目
4.  [GVP](https://gitee.com/gvp) 全称是 Gitee 最有价值开源项目，是综合评定出的优秀开源项目
5.  Gitee 官方提供的使用手册 [https://gitee.com/help](https://gitee.com/help)
6.  Gitee 封面人物是一档用来展示 Gitee 会员风采的栏目 [https://gitee.com/gitee-stars/](https://gitee.com/gitee-stars/)


CHANGELOG.md
# 0.0.0 (2023-09-08)


### Features

*  新增actions ([2103a7d](https://github.com/bestHandsome/vue3-blog/commit/2103a7dc34313160b42b277a3857e7b0340ebcf4))
* 🎸 文件调整 ([4efda90](https://github.com/bestHandsome/vue3-blog/commit/4efda907b98f21357e445b536a90aa89875f6b5d))
* 🎸 新增左侧标签栏 ([c883ba0](https://github.com/bestHandsome/vue3-blog/commit/c883ba0521a995bc8f0fd5b3cbedf6e8e008c000))
* 🎸 新增changelog ([b0eff8f](https://github.com/bestHandsome/vue3-blog/commit/b0eff8f9455de0ae7e0126eb33f7c3f7f8c8cd1c))
* 测试代码规范 ([42d9cc7](https://github.com/bestHandsome/vue3-blog/commit/42d9cc7d7d98e9fca598e28c62df756d40bc732c))
* 测试修改 ([8b26a39](https://github.com/bestHandsome/vue3-blog/commit/8b26a391c1a37ddfc4042d86aa43f584d55b5f99))
* 测试github actions ([2bdc154](https://github.com/bestHandsome/vue3-blog/commit/2bdc1544e1516686282112eae1e7bcc19177a294))
* 测试workflow ([6ba9d82](https://github.com/bestHandsome/vue3-blog/commit/6ba9d827619eda2e29cde44e1aa6262c966a08de))
* 测试workflow ([972659c](https://github.com/bestHandsome/vue3-blog/commit/972659c8d3031a8442d45d6cb0dfa42e79b7d619))
* 打包 ([f9576bc](https://github.com/bestHandsome/vue3-blog/commit/f9576bc8e4cd2a14658b250c9ea985e0b6b6c6f8))
* 打包 ([6eb7d87](https://github.com/bestHandsome/vue3-blog/commit/6eb7d87e97940b2b2e49d4d5c5f055355b8dae42))
* 打包部署 ([8bfcee1](https://github.com/bestHandsome/vue3-blog/commit/8bfcee118bb5a124ea99cce940d9fefccb6285ba))
* 配置调整 ([b274c23](https://github.com/bestHandsome/vue3-blog/commit/b274c23c5853283d78f6b551e650b353dbb91e5a))
* 文件修改 ([39f8fea](https://github.com/bestHandsome/vue3-blog/commit/39f8fea8a404d6e69fad342067deb39768ec7a6d))
* 新增文件 ([dfd98a5](https://github.com/bestHandsome/vue3-blog/commit/dfd98a59f58952f3ca0a0add744a8f4b76183421))
* 新增项目配置 ([7128371](https://github.com/bestHandsome/vue3-blog/commit/712837153d01765b18820e050ae8b1cacd484ca0))
* 新增主题 ([8ba9007](https://github.com/bestHandsome/vue3-blog/commit/8ba9007319a2fc24be5678484fda73d7918d0e46))
* 新增changelog ([4969502](https://github.com/bestHandsome/vue3-blog/commit/4969502df72e5a43c4d511b0a84daf0712d11592))
* 新增gitnore ([8a86afa](https://github.com/bestHandsome/vue3-blog/commit/8a86afa8e440cd70fbb4cbf516026693c767f99d))
* 修改配置文件 ([2e51959](https://github.com/bestHandsome/vue3-blog/commit/2e519595e0d7de5aeda0221f6b2861ee1e0c56ca))
* 修改文件测试 ([dba863f](https://github.com/bestHandsome/vue3-blog/commit/dba863f0a1f708353ad941da9a0257891c904bbe))
* 修改app.vue ([bcdc099](https://github.com/bestHandsome/vue3-blog/commit/bcdc099cd9eedf59ff046f2d9082a9a77850aa77))
* 增加单元测试 ([20eacd6](https://github.com/bestHandsome/vue3-blog/commit/20eacd6a9a4aa9d1aec88dff7115a8639bb3223e))
* 增加commit 规范 ([af7409a](https://github.com/bestHandsome/vue3-blog/commit/af7409af9d6e4a084c09e186cb3a397123465108))
* mixed content ([9759211](https://github.com/bestHandsome/vue3-blog/commit/97592115ba8305867aeee357352ff758ac7f0f68))
* yml更新 ([d884403](https://github.com/bestHandsome/vue3-blog/commit/d884403b0c54e53a686074c475d6c60ec7352e74))



# 0.0.0 (2023-09-05)


### Features

*  新增actions ([2103a7d](https://github.com/bestHandsome/vue3-blog/commit/2103a7dc34313160b42b277a3857e7b0340ebcf4))
* 🎸 文件调整 ([4efda90](https://github.com/bestHandsome/vue3-blog/commit/4efda907b98f21357e445b536a90aa89875f6b5d))
* 🎸 新增左侧标签栏 ([c883ba0](https://github.com/bestHandsome/vue3-blog/commit/c883ba0521a995bc8f0fd5b3cbedf6e8e008c000))
* 🎸 新增changelog ([b0eff8f](https://github.com/bestHandsome/vue3-blog/commit/b0eff8f9455de0ae7e0126eb33f7c3f7f8c8cd1c))
* 测试代码规范 ([42d9cc7](https://github.com/bestHandsome/vue3-blog/commit/42d9cc7d7d98e9fca598e28c62df756d40bc732c))
* 测试修改 ([8b26a39](https://github.com/bestHandsome/vue3-blog/commit/8b26a391c1a37ddfc4042d86aa43f584d55b5f99))
* 测试github actions ([2bdc154](https://github.com/bestHandsome/vue3-blog/commit/2bdc1544e1516686282112eae1e7bcc19177a294))
* 测试workflow ([6ba9d82](https://github.com/bestHandsome/vue3-blog/commit/6ba9d827619eda2e29cde44e1aa6262c966a08de))
* 测试workflow ([972659c](https://github.com/bestHandsome/vue3-blog/commit/972659c8d3031a8442d45d6cb0dfa42e79b7d619))
* 打包 ([f9576bc](https://github.com/bestHandsome/vue3-blog/commit/f9576bc8e4cd2a14658b250c9ea985e0b6b6c6f8))
* 打包 ([6eb7d87](https://github.com/bestHandsome/vue3-blog/commit/6eb7d87e97940b2b2e49d4d5c5f055355b8dae42))
* 打包部署 ([8bfcee1](https://github.com/bestHandsome/vue3-blog/commit/8bfcee118bb5a124ea99cce940d9fefccb6285ba))
* 配置调整 ([b274c23](https://github.com/bestHandsome/vue3-blog/commit/b274c23c5853283d78f6b551e650b353dbb91e5a))
* 文件修改 ([39f8fea](https://github.com/bestHandsome/vue3-blog/commit/39f8fea8a404d6e69fad342067deb39768ec7a6d))
* 新增文件 ([dfd98a5](https://github.com/bestHandsome/vue3-blog/commit/dfd98a59f58952f3ca0a0add744a8f4b76183421))
* 新增项目配置 ([7128371](https://github.com/bestHandsome/vue3-blog/commit/712837153d01765b18820e050ae8b1cacd484ca0))
* 新增主题 ([8ba9007](https://github.com/bestHandsome/vue3-blog/commit/8ba9007319a2fc24be5678484fda73d7918d0e46))
* 新增changelog ([4969502](https://github.com/bestHandsome/vue3-blog/commit/4969502df72e5a43c4d511b0a84daf0712d11592))
* 新增gitnore ([8a86afa](https://github.com/bestHandsome/vue3-blog/commit/8a86afa8e440cd70fbb4cbf516026693c767f99d))
* 修改配置文件 ([2e51959](https://github.com/bestHandsome/vue3-blog/commit/2e519595e0d7de5aeda0221f6b2861ee1e0c56ca))
* 修改文件测试 ([dba863f](https://github.com/bestHandsome/vue3-blog/commit/dba863f0a1f708353ad941da9a0257891c904bbe))
* 修改app.vue ([bcdc099](https://github.com/bestHandsome/vue3-blog/commit/bcdc099cd9eedf59ff046f2d9082a9a77850aa77))
* 增加单元测试 ([20eacd6](https://github.com/bestHandsome/vue3-blog/commit/20eacd6a9a4aa9d1aec88dff7115a8639bb3223e))
* 增加commit 规范 ([af7409a](https://github.com/bestHandsome/vue3-blog/commit/af7409af9d6e4a084c09e186cb3a397123465108))
* mixed content ([9759211](https://github.com/bestHandsome/vue3-blog/commit/97592115ba8305867aeee357352ff758ac7f0f68))
* yml更新 ([d884403](https://github.com/bestHandsome/vue3-blog/commit/d884403b0c54e53a686074c475d6c60ec7352e74))



# 0.0.0 (2023-09-05)



vite.config.ts
/*
 * @Author: xingjin
 * @Date: 2023-07-21 14:33:55
 * @LastEditors: xingjinjin
 * @LastEditTime: 2023-09-04 10:07:26
 * @Description: 请填写简介
 */
/// <reference types="vitest"/>
import { defineConfig } from 'vite';
import vue from '@vitejs/plugin-vue';
import AutoImport from 'unplugin-auto-import/vite';
import Components from 'unplugin-vue-components/vite';
import { NaiveUiResolver } from 'unplugin-vue-components/resolvers';
import VueDevTools from 'vite-plugin-vue-devtools';
// import eslint from 'vite-plugin-eslint';
import { fileURLToPath, URL } from 'node:url';

// https://vitejs.dev/config/
export default defineConfig({
	base: '/vue3-blog',
	plugins: [
		vue(), // 自动导入API方法
		VueDevTools(),
		// eslint(),
		AutoImport({
			imports: [
				'vue',
				'vue-router',
				'vitest',
				'pinia',
				{
					'naive-ui': ['useDialog', 'useMessage', 'useNotification', 'useLoadingBar'],
				},
			],
			eslintrc: {
				enabled: true, // Default `false`
				filepath: './.eslintrc-auto-import.json', // Default `./.eslintrc-auto-import.json`
				globalsPropValue: true, // Default `true`, (true | false | 'readonly' | 'readable' | 'writable' | 'writeable')
			},
			resolvers: [], // custom resolvers
			dts: 'typings/auto-imports.d.ts', // 导入存放地址
		}),
		// 自动导入组件
		Components({
			resolvers: [NaiveUiResolver()], // custom resolvers
			dts: 'typings/components.d.ts',
		}),
	],
	resolve: {
		alias: {
			'@/': fileURLToPath(new URL('./src/', import.meta.url)),
		},
	},
	test: {
		globals: true,
		environment: 'jsdom',
		coverage: {
			provider: 'v8',
		},
	},
	server: {
		host: '0.0.0.0',
		hmr: true,
		open: true,
		proxy: {
			'/api': {
				target: 'http://172.16.88.41:1337',
				changeOrigin: true,
				secure: false,
				// rewrite: path => path.replace(/^\/api/, ''),
			},
			'/uploads': {
				target: 'http://172.16.88.41:1337',
				changeOrigin: true,
				secure: false,
			},
		},
	},
});
